android/src/main/java/ie/gov/tracing/network/DiagnosisKeyDownloader.kt

```diff
@@ -35,8 +35,10 @@ internal class DiagnosisKeyDownloader(private val context: Context) {
         }        
         if (since <= 0) {
             val startIndex = max(0, files.size - fileLimit.toInt())
-            return files.subList(startIndex, fileLimit.toInt()).toTypedArray()
+            val endIndex = min(files.size, startIndex + fileLimit.toInt())
+            return files.subList(startIndex, endIndex).toTypedArray()
         } else {
+            val endIndex = min(files.size, fileLimit.toInt())
             return files.subList(0, fileLimit.toInt()).toTypedArray()
```

Comment:
`endIndex` appears to be unused here (and could be out of bounds as above)?

Reply:
Will fix this